### PR TITLE
Add human-regularized LBF policy training

### PR DESCRIPTION
**Pull request by Montekkundan**
Adds the local LBF human-regularized policy code that was sitting outside the PR.

Includes:
- BC feature augmentation and checkpoint selection helpers
- HR-IPPO/reference-KL sweep/eval scripts
- human-reg config knobs for PPO ego/IPPO

Tests:
- .venv/bin/python -m compileall agents/bc/bc_lstm.py agents/bc/evaluate_lbf.py agents/bc/evaluate_lbf_dataset_partners.py agents/bc/train.py agents/lbf/__init__.py agents/lbf/agent_policy_wrappers.py ego_agent_training/ppo_ego.py marl/ippo.py teammate_generation/train_ego.py agents/bc/evaluate_ppo_human_likeness.py agents/bc/lbf_features.py agents/bc/run_hr_ippo_lbf_sweep.py agents/bc/run_human_reg_ppo_sweep.py agents/bc/select_lbf_checkpoint_by_return.py agents/bc/summarize_human_proxy_results.py tests/test_lbf_human_reg_features.py
- direct LBF feature augmentation check

Note: result CSVs/checkpoints are left out of git.

### CODE REVIEW

File: agents/bc/__init__.py
Comment (carolinewang01) [inline agents/bc/__init__.py:1]: Can you reorganize the code inside `agents/bc` so that: 
- the actual human_reg_rl training/experiment scripts are moved to a new directory within `scripts/`?
- the infrastructure for initializing the LBF BC policies are moved to agents/lbf/bc, since it looks lbf specific?

File: agents/bc/bc_lstm.py [hunk 1]
```diff
@@ -117,26 +119,108 @@ def _flatten(prefix, d):
...
         _flatten("", self.params)
-        os.makedirs(os.path.dirname(path), exist_ok=True)
+        output_dir = os.path.dirname(path)
+        if output_dir:
+            os.makedirs(output_dir, exist_ok=True)
         save_file(flat_dict, path)
 
 
-def compute_bc_loss(params, network, carry, obs_seq, action_seq, avail_seq, mask):
+class BCLSTMPolicyWrapper(AgentPolicy):
```
Comment (carolinewang01) [inline agents/bc/bc_lstm.py]: should be moved to agents/lbf/agent_policy_wrappers.py

File: agents/bc/bc_lstm.py [hunk 2]
```diff
@@ -117,26 +119,108 @@ def _flatten(prefix, d):
...
+        pi = distrax.Categorical(logits=logits)
+        action = jax.lax.cond(
+            test_mode,
+            lambda: pi.mode(),
+            lambda: pi.sample(seed=rng),
+        )
+        return action.reshape(obs_shape), hstate
+
+
+def compute_bc_loss(params, network, carry, obs_seq, action_seq, avail_seq, mask,
```
Comment (carolinewang01) [inline agents/bc/bc_lstm.py]: where is this used? if you have a script to run BC training, it would be nice to check it into `scripts/bc`. Then this loss computation function should probably be moved there.

File: agents/bc/__init__.py
Comment (carolinewang01) [reply agents/bc/__init__.py:1]: the agents folder is meant to contain only classes directly implementing agents / agent neural architectures.

**Review by carolinewang01 — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 3]
```diff
@@ -1,17 +1,25 @@
 training_state
 figures/
+figures/*
 notebooks/
+notebooks/*
```
Comment (carolinewang01) [inline .gitignore]: These seem redundant with the current gitignored folders. Please fix.

File: scripts/bc/lbf_features.py
Comment (carolinewang01) [inline scripts/bc/lbf_features.py:1]: Since these features are only for the BC, can we move it to the scripts/bc dir?

File: human_data_processing/load_lbf_data.py [hunk 4]
```diff
@@ -215,6 +215,12 @@ def load_bc_data_padded(config_name: str,
     )
 
 
+# Backward-compatible names used by human_data_processing/README.md.
```
Comment (carolinewang01) [inline human_data_processing/load_lbf_data.py]: Wait, is `load_bc_data` one of the old names? That should be a new name introduced by this PR right? In any case, do we need backwards compatibility? It seems a bit messy.

File: ego_agent_training/ppo_ego.py [hunk 5]
```diff
@@ -37,6 +38,100 @@
 logging.basicConfig(level=logging.INFO)
 
 
+class GAILDiscriminator(nn.Module):
```
Comment (carolinewang01) [inline ego_agent_training/ppo_ego.py]: Was this used in the experiments that ended up in the paper? If not, perhaps we should discard the changes to the ego agent training script.

File: marl/ippo.py [hunk 6]
```diff
@@ -24,6 +24,40 @@
 from marl.ppo_utils import Transition, batchify, unbatchify, _create_minibatches
 
 
+def _load_human_reference_policy(config):
```
Comment (carolinewang01) [inline marl/ippo.py:27]: I get why you modified marl/ippo.py to implement human-regularized self-play, but these changes might be too messy to merge into experiments.

**Review by carolinewang01 — CHANGES_REQUESTED** [review]
Thanks for your patience. I made a couple of minor requests around organization and asked a few questions, but overall I think the feature is too extensive to go into expts branch. Can you fix the small stuff, document how the features in this branch were used in the paper, and rename this branch expts/human_reg_lbf? Then we can leave this code there.

File: .gitignore (hunk 3 above)
Comment (Montekkundan) [reply .gitignore]: Removed the redundant ignore patterns.

File: scripts/bc/lbf_features.py
Comment (Montekkundan) [reply scripts/bc/lbf_features.py:1]: Moved this to scripts/bc/lbf_features.py and updated the imports.

File: human_data_processing/load_lbf_data.py (hunk 4 above)
Comment (Montekkundan) [reply human_data_processing/load_lbf_data.py]: I removed the aliases and updated the README to use the load_bc_data* names directly.

File: ego_agent_training/ppo_ego.py (hunk 5 above)
Comment (Montekkundan) [reply ego_agent_training/ppo_ego.py]: This path was not used in the paper experiments, so I removed the ego-PPO/GAIL changes and the corresponding sweep script.

File: marl/ippo.py (hunk 6 above)
Comment (Montekkundan) [reply marl/ippo.py:27]: I kept it only on the archival branch and documented it in `scripts/bc/README.md`; will not be merged into `expts`
